### PR TITLE
Updates to login docs (login step, Entra branding)

### LINKED ISSUE #1571: Should `AAD` be changed to `Azure Entra ID`?

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR was originally made to clarify that you must login to your auth tenant if its different from main tenant. But then I also fixed the Markdown lint issues and updated all Azure AD mentions.

Fixes #1571

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

N/A

### CODE REVIEW

File: docs/login_and_acl.md [hunk 1]
```diff
@@ -124,18 +124,19 @@ Consent from the user must be obtained for use of the client and server app. The
...
 
-### Troubleshooting Azure AD Setup
+### Troubleshooting
 
-* If any Azure AD apps need to be recreated, you can avoid redeploying the app by [changing the app settings in the portal](https://learn.microsoft.com/azure/app-service/configure-common?tabs=portal#configure-app-settings). Any of the [required environment variables](#environment-variables-reference) can be changed. Once the environment variables have been changed, restart the web app.
+* If your primary tenant restricts the ability to create Entra applications, you'll need to use a separate tenant to create the Entra applications. You can create a new tenant by following [these instructions](https://learn.microsoft.com/entra/identity-platform/quickstart-create-new-tenant). Then run `azd env set AZURE_AUTH_TENANT_ID <YOUR-AUTH-TENANT-ID>` before running `azd up`.
```
Comment (pamelafox) [inline docs/login_and_acl.md:135]: Added this new one, please review.

File: docs/login_and_acl.md [hunk 2]
```diff
@@ -39,24 +39,25 @@ The easiest way to setup the two apps is to use the `azd` CLI. We've written scr
 1. Ensure access control is enabled on your search index. If your index doesn't exist yet, run prepdocs with `AZURE_USE_AUTHENTICATION` set to `true`. If your index already exists, run `pwsh ./scripts/manageacl.ps1 --acl-action enable_acls`.
 1. (Optional) To require access control when using the app, run `azd env set AZURE_ENFORCE_ACCESS_CONTROL true`.
 1. Run `azd env set AZURE_AUTH_TENANT_ID <YOUR-TENANT-ID>` to set the tenant ID associated with authentication.
-2. Run `azd up` to deploy the app.
+1. If your auth tenant ID is different from your currently logged in tenant ID, run `azd auth login --tenant-id <YOUR-TENANT-ID>` to login to the authentication tenant simultaneously.
```
Comment (pamelafox) [inline docs/login_and_acl.md:42]: New step

**Comment by pamelafox** [comment]
CI checks passing now, ready for review.

**Comment by pamelafox** [comment]
@mattgotteiner Ready for review :)

**Review by mattgotteiner — APPROVED** [review]

**Pull request merged**